README.md

```diff
@@ -381,464 +509,224 @@ class MyBaseClass extends WidgetBase<WidgetProperties> {
-.root {
-	background: red;
-}
-
-.tab {
-	background: blue;
-}
-```
-
-##### Registering `baseClasses`
-
-To apply `baseClasses` a widget must use `ThemeableMixin` and the `theme` decorator from `mixins/Themeable` to register the `baseClasses`.
-
-```ts
-import { WidgetProperties } from '@dojo/widget-core/interfaces';
-import { WidgetBase } from '@dojo/widget-core/WidgetBase';
-import { ThemeableMixin, theme } from '@dojo/widget-core/mixins/Themeable';
-import * as baseClasses from './styles/tabpanel.css';
-
-@theme(baseClasses)
-class MyThemeableWidget extends ThemeableMixin(WidgetBase)<WidgetProperties> {
-	// ...
-}
-```
-
-Basic usage:
-
-```ts
-import { WidgetProperties } from '@dojo/widget-core/interfaces';
-import { WidgetBase } from '@dojo/widget-core/WidgetBase';
-import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
-import * as baseClasses from './styles/tabpanel.css';
-
-interface MyThemeableWidgetProperties extends WidgetProperties, ThemeableProperties;
-
-@theme(baseClasses)
-class MyThemeableWidget extends ThemeableMixin(WidgetBase)<MyThemeableWidgetProperties> {
-	render: function () {
-		const { root, tab } = baseClasses;
-		return
-			v('ul', { classes: this.classes(root) }, [
-				v('li', { classes: this.classes(tab) }, [ 'tab1' ])
-				// ...
-			]);
-	}
-}
-```
-
-##### Applying a Theme
-
-Themeable widgets include an optional `theme` property which can be set to pass in a theme. Theme classes will override `baseClasses`. When a `theme` property is set or changed, the widgets `theme` classes will be regenerated and the widget invalidated such that it is redrawn. Themes are used to apply consistent styling across the widget codebase.
-
-Usage Extending on the previous `tabPanel` example.
+```typescript
+import MetaBase from "@dojo/widget-core/meta/Base";
 
-``` css
-/* customTheme/tabPanel.css */
-.tabs {
-	background: green;
+interface IsTallMetaOptions {
+    minHeight: number;
 }
-```
-
-Import the theme and pass it to the widget via its `properties`. The theme classes will be automatically mixed into the widget and available via `this.classes`.
-
-```ts
-import * as customTheme from './themes/customTheme.css';
-
-interface MyThemeableWidgetProperties extends WidgetProperties, ThemeableProperties;
 
-class MyThemeableWidget extends ThemeableMixin(WidgetBase)<MyThemeableWidgetProperties> {
-	render: function () {
-		// Resulting widget will have green tabs instead of baseTheme red.
-		return w(TabPanel, { theme: customTheme });
-	}
+class IsTallMeta extends MetaBase {
+    isTall(key: string, { minHeight }: IsTallMetaOptions = { minHeight: 300 }): boolean {
+        this.requireNode(key);
+        const node = this.nodes.get(key);
+        if (node) {
+            return node.offsetHeight >= minHeight;
+        }
+        return false;
+    }
 }
 ```
 
-The theme can be applied to individual widgets or to a project and property passed down to its children.
+### Tsx Support
```

Comment:
`TSX support`? JSX always appears all-caps: https://facebook.github.io/react/docs/jsx-in-depth.html and https://www.typescriptlang.org/docs/handbook/jsx.html, also even TypeScript seems to just refer to it as JSX.